pkg/fanal/types/package.go

```diff
@@ -57,6 +57,7 @@ type PkgIdentifier struct {
 	UID    string                 `json:",omitempty"` // Calculated by the package struct
 	PURL   *packageurl.PackageURL `json:"-"`
 	BOMRef string                 `json:",omitempty"` // For CycloneDX
+	SPDXID string                 `json:",omitempty"` // For SPDX
```

Comment:
@knqyf263 wdyt about using `BOMRef` for `SPDXID`?
Name (I mean `BOMRef`) can be confusing, but using two fields for similar fields is also wrong

Reply:
I thought about it again.
Using two same fields doesn't make sense and may lead to bugs.
So i used `PkgIdentifier.BOMRef` for `SPDXID` in https://github.com/aquasecurity/trivy/pull/6907/commits/964f16aa28c806d9a2f3f82d97e4ab81414fa9d8.

Reply:
How about the case where the input is CycloneDX, and the output is SPDX? I'm concerned BOM-Ref is mistakenly used as SPDX-ID, as we are currently trying to reuse the same ID after https://github.com/aquasecurity/trivy/pull/7340.

Reply:
you are right. After #7340 it can be problem.
I thought about adding `SPDXID` field (https://github.com/aquasecurity/trivy/pull/6907/commits/68abc7802a7e60495c0197a82a1f747c5765341a)
But it looks as bad way.

What if we will use struct with type + ref/ID. e.g.:
```go
	type BomID struct {
		ID   string // UUID/purl/SPDXID
		Type BomType // cyclonedx/spdx
	}
```
If BomID.Type == outputType => save ID, otherwise - create new ID.